source/MissionAction.h

```diff
@@ -83,7 +84,8 @@ class MissionAction {
 	Conversation conversation;
 	
 	std::map<const GameEvent *, std::pair<int, int>> events;
-	std::map<const Outfit *, int> gifts;
+	std::vector<std::pair<std::string, std::string>> giftShips;
```

Comment:
giftShips should be a map like giftOutfits. I'd suggest `std::map<const Ship *, std::string>` where the string is the name. Reason being that as this code is written right now, a mission could be written to gift a nonexistent ship (e.g. the mission writer had a typo in the ship name), but this won't throw an error until the mission actually tried to gift the ship, because that is the only point where the string for the ship model is actually converted into the actual ship model.